sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/HistogramPointData.java

```diff
@@ -28,6 +29,22 @@
    */
   long getCount();
 
+  /**
+   * The min of all measurements recorded.
+   *
+   * @return the min of recorded measurements, or {@code null} if not available.
+   */
+  @Nullable
```

Comment:
Arguably we should add `@Nullable` to `getSum()` as wel since its optional according to the [proto](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L462).

Although, its not (currently) optional in the [data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md?plain=1#L499). Wondering if that's an oversight or intentional.

Reply:
I guess we don't need to make these nullable either - we always populate it anyways, I'm not sure if we get any practical flexibility by it being nullable (i.e., we could potentially decide to not populate it in the future, but why would we?)

Reply:
I think I would treat the optional on whether an SDK supports it or not - but if we're adding support, we don't need to match that with a Nullable, we are supporting the optional feature all the time

Reply:
Also tangentially related, I brought up the issue of understanding the relationship between the proto and data model in https://github.com/open-telemetry/opentelemetry-specification/pull/2429#issuecomment-1082585856. I don't think it necessarily always needs to be the case that both match up, but there can probably be some more documentation to understand these cases better.

Reply:
There are a couple of situations where we may want to leave the min / max values unset:
- When bridging metrics from another system. For example, opencensus histograms dont have a min / max so I've set min / max to `null` in the opencensus shim.
- Taking the diff between two histograms. The asynchronous instruments are assumed to observe cumulative values, and so if we need to export in delta format, we need to take the diff. Min and max have no useful values when computing from cumulative to delta. Practically speaking, I don't think it would make sense to configure the aggregation to be histogram for any of the current async instruments instruments  (and in fact, the current output when you do such a configuration is surprising and arguably wrong). Maybe in the future there might be other observable instruments that do make more sense as a histogram, and for which a diff is necessary. 

Along similar lines, if we wanted to be be able to offer any sort of open metrics bridge, or in general support metrics systems that allow negative recordings for histograms, we'd need some way to indicate that `sum` is not set, else we break the monotonicity assumption of `sum`.

Reply:
We discussed in the 3/31 Office Hours using `hasMin()` and `hasMax()` instead of `@Nullable`. I prefer this and have added a commit to switch to this strategy. However, I've kept `@Nullable` on the internal `HistogramAccumulation` `getMin` and `getMax` methods because its makes the code a bit lit verbose. Don't feel strongly about it.

Reply:
I've been giving more thought to whether or not we should add `boolean hasSum()` to reflect that `sum` is required in the data model, but optional in the [proto](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L462), and I don't think its necessary:

- `hasSum()` would be false if bridging a histogram from a metrics system that allows negative measurements. We don't have this bridging use case today, and if we did, we could add a new `default boolean hasSum() { return true; }` and be ok.
- In the future, otel histograms instruments may support negative measurements, but for now that's an unsolved problem and sum is monotonic. If histograms change to support negative measurements, there will be need to be some sort of indication that sum is non-monotonic. In this case, we wouldn't want a `hasSum()` method since sum would always be present, but a `isMonotonic()` method. 

So we have a reasonable path forward in both scenarios without having to solve it now.